parser/ParseStatement.hpp

```diff
@@ -258,6 +292,22 @@ class ParseStatementCreateIndex : public ParseStatement {
       return index_type_.get();
     }
 
+    /**
+     * @brief Get the index properties associated with this index type.
+     *
+     * @return The index properties for this type.
+     **/
+    const IndexProperties* getIndexProperties() const {
+      return index_properties_.get();
+    }
+
+    const ParseIndexProperties* getCustomPropertiesNode() const {
+      return custom_properties_node_.get();
```

Comment:
For above two functions, prefer to return the reference, unless they could be `NULL`.